### PR TITLE
[UBS.Admin.Permissions] API: Admin is able to edit SuperAdmin's permissions #5554

### PR DESCRIPTION
**Pull request by ABbondar**
# GreenCityUBS PR
https://github.com/ita-social-projects/GreenCity/issues/5554

## Summary Of Changes :fire:
Added validation for updating permissions for employee.
Added several new endpoints and tests.

## Issue Link :clipboard:
_&lt;Link to the issue&gt;_

## Added
* Added validation for updating employee's permissions.
* Added new endpoint for getting information about login employee`s positions.
* Added new endpoint for geting information about an employee`s positions and all possible related authorities to these positions.
* Added ErrorDecoder for Feign Exceptions.

## Changed
* _&lt;detail item of what was changed&gt;_

## Deleted
* _&lt;detail item of what was removed&gt;_

## How to test :clipboard:
In can be test via swagger:
[/admin/ubs-employee/edit-authorities]
[/admin/ubs-employee/get-employee-login-positions]
[/admin/ubs-employee/get-positions-authorities]

# PR Checklist Forms

_(to be filled out by PR submitter)_
- [ ] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-XXXX: Ticket title)
- [x] This template filled (above this section).
- [x] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells ar duplication
- [x] `NEED_REVIEW` and `READY_FOR_REVIEW` labels are added.
- [x] All files reviewed before sending to reviewers

### CODE REVIEW

File: service-api/src/main/java/greencity/client/UserRemoteClient.java [hunk 1]
```diff
@@ -65,6 +71,27 @@ public interface UserRemoteClient {
     @GetMapping("/user/checkByUuid")
     boolean checkIfUserExistsByUuid(@RequestParam(UUID) String uuid);
 
+    /**
+     * Gets current user's positions and all possible related authorities to these
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/client/UserRemoteClient.java]: it returns positions for user with `email` not for the current user

Comment (ABbondar) [reply service-api/src/main/java/greencity/client/UserRemoteClient.java]: edited

File: core/src/main/java/greencity/exception/handler/RetrieveMessageErrorDecoder.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+@Component
+public class RetrieveMessageErrorDecoder implements ErrorDecoder {
+    @Override
+    public Exception decode(String methodKey, Response response) {
+        Map<String, Object> errorAttributes = new HashMap<>();
+        errorAttributes.put("message", "An error occurred");
+        errorAttributes.put("timestamp", "");
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/exception/handler/RetrieveMessageErrorDecoder.java]: why do you use an empty timestamp?

Comment (RomaMocherniuk) [reply core/src/main/java/greencity/exception/handler/RetrieveMessageErrorDecoder.java]: please move `message` and `timestamp` to the constants

Comment (ABbondar) [reply core/src/main/java/greencity/exception/handler/RetrieveMessageErrorDecoder.java]: edited

**Review by RomaMocherniuk — APPROVED** [review]

**Review by RomaMocherniuk — APPROVED** [review]

**Review by sergoliarnik — APPROVED** [review]

**Review by juseti — APPROVED** [review]

**Review by MaksymGolik — APPROVED** [review]

**Review by LiliaMokhnatska — APPROVED** [review]

**Pull request merged**